x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -656,7 +674,7 @@ private static IndicesPrivilegesWithOptionalRemoteClusters parseIndexWithOptiona
         String currentFieldName = null;
         String[] names = null;
         BytesReference query = null;
-        String[] privileges = null;
+        String[] privileges = predefinedPrivileges;
```

Comment:
Nit: we could assert that `privileges` is null or not empty, to fail early.

Reply:
The IndicesPrivileges.Builder does check for `null` or empty privileges. That said, an extra assertion does not hurt either. So I added one.